crates/shared/src/sources/balancer_v2/pools.rs

```diff
@@ -100,6 +102,10 @@ pub trait FactoryIndexing: Send + Sync + 'static {
     /// Additionally, a block spec and a batch call context is passed in to
     /// specify exactly the block number the state should be read for, and allow
     /// for more optimal performance when fetching a large number of pools.
+    ///
+    /// This method should return `None` if the pool is disabled. This allows
+    /// pool/factory specific details about whether or not the pool can be
+    /// used to be provided to the caller.
     fn fetch_pool_state(
```

Comment:
I guess this means that we short circuit the web3 calls if we find the pool in paused state. As in here is not reason for use to fetch the reserves and serves as a nice optimization! Very cool.